File: typescript/src/data_models/process_model/process_model.ts
import {Event} from '../event/index';

/**
 * Describes a ProcessModel.
 * The content is always user-specific and will not contain any information that a requesting user is not authorized to see.
 */
export class ProcessModel {
  /**
   * The id of the ProcessModel.
   */
  public id: string;
  /**
   * Contains a list of StartEvents that the requesting user has access to.
   */
  public startEvents: Array<Event> = [];
  /**
   * Contains a list of EndEvents that the requesting user has access to.
   */
  public endEvents: Array<Event> = [];
}

File: dotnet/src/data_models/process_model/process_model.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    using System.Collections.Generic;

    /// <summary>
    /// Describes a ProcessModel.
    /// </summary>
    public class ProcessModel
    {
        /// <summary>
        /// The ProcessModel's ID.
        /// </summary>
        public string ID { get; set; }

        /// <summary>
        /// A list of accessible StartEvents provided by the ProcessModel.
        /// </summary>
        public List<Event> StartEvents { get; set; }

        /// <summary>
        /// A list of accessible EndEvents provided by the ProcessModel.
        /// </summary>
        public List<Event> EndEvents { get; set; }
    }
}
